### PR TITLE
Update the procedures primer for better viewing in the online documentation

### PR DESCRIPTION
**Pull request by lydia-duncan**
Adds some headers, clarifies some of the descriptions, encloses some code
snippets in backticks.

Verified that the primer still passes and that I like the generated html.

### CODE REVIEW

File: test/release/examples/primers/procedures.chpl [hunk 1]
```diff
@@ -233,12 +258,12 @@ writeln("bip is now ", bip);	// 0
...
+// The ``out`` intent causes the actual argument to be ignored when the
+// procedure starts. The actual is assigned the value of the corresponding
+// formal when the routine exits.
 //
-// This arctan routine puts the result in res and returns the number of
-// iterations it needed to converge.
+// This arctan routine puts the result in the argument with ``out`` intent
+// and returns the number of iterations it needed to converge.
 //
 // atan x = x - x^3/3 + x^5/5 + sum_3^inf (-1)^i x^(2i+1)/(2i+1).
```
Comment (ben-albrecht) [inline test/release/examples/primers/procedures.chpl]: Can you inline code this? 

e.g. ```` ``atan x = x - x^3/3 + x^5/5 + sum_3^inf (-1)^i x^(2i+1)/(2i+1).`` ````

File: test/release/examples/primers/procedures.chpl [hunk 2]
```diff
@@ -106,13 +112,27 @@ writeln("down + over = ", down + over);
...
   var center : Point;
   var radius : real;
 }
 
+//
+// Note that a default value for an argument can be provided,
+// which will be used if a value for that argument is not
+// specified in the call.  Here, instead of specifying the
+// type of ``x``, ``y``, and ``diameter``, we provide them a default
+// value of ``0.0``.  Because we did not specify their type
```
Comment (lydia-duncan) [inline test/release/examples/primers/procedures.chpl]: Ugh, good point.  I think I'll add back tics to that line you pointed out.

**Comment by ben-albrecht** [comment]
GitHub won't let me line comment sections that you didn't edit, but for [line 276](https://github.com/chapel-lang/chapel/pull/4589/files#diff-9f9822e10e6079817cb39c01ff24d820L276), you should be able to add a link to `varargs.chpl` with:

``` :ref:`primers-varargs` ```

Note: you'll have to be working on top of the changes from #4588 for this label to exist.

**Comment by ben-albrecht** [comment]
Huge improvement. Nice work - good to merge after my comments.

**Pull request merged**